articles/nginx-reverse-proxy/index.md

```diff
@@ -0,0 +1,192 @@
+
+### Step 3 - Configure NGINX at 80 for HTTP and 443 for HTTPS
+Now that we have our apps running and our DNS records ready. We can start configuring our NGINX Reverse Proxy to make it all work.  
+
+By default, the configuration file is named nginx.conf and placed in the directory /usr/local/nginx/conf, /etc/nginx, or /usr/local/etc/nginx for Linux and Debian Based systems.
+
+On Windows, the file is placed inside the installation folder, nginx/conf/nginx.conf.
+
+Add these configurations inside the HTTP block.
+
+#### Step 3.1 - HTTP
+
+```
+server {       
+     listen       80;
+     server_name  server.domain;
+     location / {
+          proxy_pass "http://localhost:3000" ;
+     }
+}
+server {       
+     listen       80;
+     server_name  client.domain;
+
+     location / {
+          proxy_pass "http://localhost:3001" ;
+     }
+}
+```
+
+#### Step 3.2 - HTTPS
+
+```
+server {
+    listen 443 ssl http2;
+    listen [::]:443 ssl http2;
+
+    server_name server.domain;
+
+    location / {    
+        proxy_pass "http://localhost:3000/";
+    }
+
+    ssl_certificate <location of SSL certificate>;
+    ssl_certificate_key <location of SSL certificate Key>;
+}
+server {
+    listen 443 ssl http2;
+    listen [::]:443 ssl http2;
+
+    server_name client.domain;
+
+    location / {    
+        proxy_pass "http://localhost:3001/";
+    }
+
+    ssl_certificate <location of SSL certificate>;
+    ssl_certificate_key <location of SSL certificate Key>;
+}
+```
+
+**NOTE**
+These are the basic minimum configurations required to successfully configure NGINX for reverse proxying. Feel free to explore other config parameters as well.
+
+Make sure to change the domain name to your domain. For a SSL Certificate and Key, you can obtain them from your SSL provider.
+
+If you don't have one, Use this free service [LetsEncrypt](https://letsencrypt.org/).
+
+Follow their documentation to get Free SSL instantly!
+
+### Step 4 - Save and Restart
+After editing, save your changes. Use the `nginx -t` command to test your changes before actually reloading NGINX. It is good practice do this to make sure your server won't crash if there were any errors in your config file.
+
+Once you get a message that the test is successful, you can go ahead and restart NGINX.  
+Open the browser and enter the URLs to find your applications running on the corresponding URLs configured.
+
+**Important Note**
+> Using NGINX secures your server because it routes the traffic internally. Instead of having to open up all of your ports, in this case 3000 and 3001, to the internet, just 80 and 443 will do the trick.  
```

Comment:
It would be nice if you listed the URLs based on the example you provided.

Reply:
Thanks and Done 👍